File: bindings/typescript/examples/upbit-provider.mjs
// Read Upbit Korea provider-specific quotation data.
import { Exchange, Market, UpbitAdapter, initialize } from "@jabdori/maxt/node";

await initialize();

const adapter = new UpbitAdapter();
const market = Market.spot(Exchange.Upbit, "BTC", "KRW");
const [tickers, instruments] = await Promise.all([
  adapter.tickers([market]),
  adapter.orderbookInstruments([market]),
]);
console.log(`region=${adapter.region.id}; ticker rows=${tickers.length}`);
console.log(`order-book instrument rows=${instruments.length}`);

File: examples/upbit_provider.rs
//! Read several Upbit Korea quotation APIs in one program.
//!
//! ```text
//! cargo run --example upbit_provider
//! ```

use maxt::adapters::UpbitAdapter;
use maxt::{Exchange, Market};

#[tokio::main(flavor = "current_thread")]
async fn main() -> maxt::Result<()> {
    let adapter = UpbitAdapter::new();
    let market = Market::spot(Exchange::Upbit, "BTC", "KRW");
    let tickers = adapter.tickers(std::slice::from_ref(&market)).await?;
    let instruments = adapter
        .orderbook_instruments(std::slice::from_ref(&market))
        .await?;
    println!(
        "region={:?}, ticker rows={}",
        adapter.region(),
        tickers.len()
    );
    println!("order-book instrument rows={}", instruments.len());
    Ok(())
}
